### ISSUE TITLE
Getting first value from Microsoft.Xrm.Sdk.EntityCollection

### ISSUE THREAD

**Issue by andreaPorini** [DigitalFlow/Xrm-Templating-Language#11]
Hi DigitalFlow,

this is a wonderful tool but i have a question: i need to create a template for an email and i want to insert the name of the first entity in the "to" recipient.

When i write ${{Value("to")}} in my template, i get "Microsoft.Xrm.Sdk.EntityCollection" as resolution for the formula; i tried to extract the first entity in the collection but i'm not able to do it.

Can you please help me?

Regards,
Andrea

**Comment by DigitalFlow** [comment]
Hi @andreaPorini,

I'm glad that you like it.
Usually there is the First() function, but it is not working on EntityCollections right now.
I'll adapt it so that you're able to use it for this. In addition to that, I think the default string representation for entity collections should be a list of all primary attribute values for the records contained.

I'll notify you once this is implemented.

Kind Regards,
Florian

**Comment by DigitalFlow** [comment]
Implemented in release v3.1.0, please test.

``` Javascript
Value ( "partyid.fullname", { explicitTarget: First( Value("to") ) } )
```

**Comment by andreaPorini** [comment]
Thank you Florian, amazing speed!

The function works correctly and it's very easy to use and understand, great!

I'll keep in touch, great job and great tool!

Kind regards,
Andrea
